### PR TITLE
gemini: postOnly and stop-limit orders

### PR DESCRIPTION
**Pull request by DoctorSlimm**

### CODE REVIEW

File: js/gemini.js [hunk 1]
```diff
@@ -991,6 +1006,88 @@ module.exports = class gemini extends Exchange {
...
+        //          "options":[],
+        //          "price":"2800.00",
+        //          "original_amount":"0.014434",
+        //          "remaining_amount":"0"
+        //      }
+        //
+        return this.parseOrder (response);
+    }
+
+    async createOrder2 (symbol, type, side, amount, price = undefined, params = {}) {
```
Comment (kroitor) [inline js/gemini.js:1090]: We will probably need to remove createOrder2, I assume it's a leftover after testing.

Comment (DoctorSlimm) [reply js/gemini.js:1090]: I am so sorry, good spot, thanks

File: js/gemini.js [hunk 2]
```diff
@@ -991,22 +1006,60 @@ module.exports = class gemini extends Exchange {
...
+                } else if ((timeInForce === 'FOK') || (timeInForce === 'fill-or-kill')) {
+                    request['options'] = [ 'fill-or-kill' ];
+                } else if (timeInForce === 'PO') {
+                    request['options'] = [ 'maker-or-cancel' ];
+                }
+            }
+            const postOnly = this.safeValue (params, 'postOnly', false);
+            params = this.omit (params, 'postOnly');
+            if (postOnly) {
+                request['options'] = [ 'maker-or-cancel' ];
```
Comment (kroitor) [inline js/gemini.js:1055]: @DoctorSlimm have you tested these options in py/php? If not, please, run a few cli commands with cli.py and cli.php just to make sure just they work, cause we don't check it on travis yet.

Comment (DoctorSlimm) [reply js/gemini.js:1055]: @kroitor I already did, thats why I changed it in the `update comments and fix php postOnly` commit

Comment (DoctorSlimm) [reply js/gemini.js:1055]: @kroitor I try to make sure that I test everything in node, python, and php having learnt from past experience that it doesnt mean it works if it only works in js 😅

**Pull request merged**